File: Leetcode/520-DetectCapital.cpp
class Solution {
  static auto isCapital(const char x) -> bool { return 'A' <= x and x <= 'Z'; };

 public:
  auto detectCapitalUse(const string& word) -> bool {
    const auto num_capitals = count_if(begin(word), end(word), isCapital);

    if (word.size() == num_capitals) return true;
    if (num_capitals == 0) return true;
    if (num_capitals == 1 and isCapital(word[0])) return true;

    return false;
  }
};

File: Leetcode/520-DetectCapital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        num_capitals = sum(letter.isupper() for letter in word)
  
        if len(word) == num_capitals: return True
        if num_capitals == 0: return True
        if num_capitals == 1 and word[0].isupper(): return True

        return False